### DISCUSSION TITLE
SAE 1.6.21 Installation erfolgreich - Einbindung Shelly Plug S & Plus 1PM

### DISCUSSION THREAD

**Discussion by masterbillRE11** [camueller/SmartApplianceEnabler#368] [Installation und Betrieb]
Hello together,

zuallererst vielen Dank für dieses tolle Projekt - bei mir > Einbindung weiterer Geräte im SHM 2.0. mithilfe der nachfolgenden Produkte: 

- Shelly Plug S
- Shelly Plus 1PM 

Die Installation erfolgte automatisch über das Installationsscript auf einem Rasp3. 
SAE über Port :8080 lässt sich starten. 

Leider bekomme ich die ShellyPlug S Steckdosen, welche nach Anleitung hinzugefügt wurden nicht als steuerbare Geräte angezeigt. 

Welche Schritte habe ich möglicherweise vergessen bzw. welche mögliche Fehlerquellen sollte ich mir gezielt anschauen. 
Welche Informationen helfen euch zur Fehlerermittlung weiter? 

Herzlichen Dank für ein erstes Feedback! 

Lieben Gruß
masterbill

**Comment by camueller** [comment]
In der Übersicht werden nur Geräte angezeigt, nicht deren Schalter oder Zähler. Hast Du einen Schalter (Shelly) konfiguriert für das Gerät?
Poste doch mal einen Screenshot der Konfiguration des Gerätes und der des Schalters ...

**Reply by masterbillRE11** [reply]
Ah - ok. Habe jetzt den Verbraucher im SHM hinzufügen können. Schalter wird erkannt und ist im SHM grün. 
Leider gibt die Steckdose den aktuellen Stromverbrauch nicht an den SHM weiter.

Anbei der Screeshot des Schalters Plug S

![image](https://user-images.githubusercontent.com/88536410/196906747-42bbd979-6b0f-495a-9454-d3289768d2fa.png)

**Reply by camueller** [reply]
Kopier doch mal die URL aus dem Screenshot in einen Brower und poste hier die Response. Am Besten, wann die Leistungsaufnahme größer als 0 ist ...

**Reply by masterbillRE11** [reply]
Ergebnis = Not found. 
Über 192.168.178.74 wird aber die Oberfläche des shellplug s ordnungsgemäß angezeigt.

**Reply by camueller** [reply]
> Ergebnis = Not found. Über 192.168.178.74 wird aber die Oberfläche des shellplug s ordnungsgemäß angezeigt.

Wenn Du da im Browser nichts bekommst, wie soll dann der SAE etwas bekommen?

**Reply by masterbillRE11** [reply]
Wie kann ich denn hier Abhilfe schaffen? Der Shelly-Plug wurde ordnungsgemäß per WLAN im Heimnetz eingebunden und lässt sich über die ip-Adresse öffnen. Danke nochmal für den prompten Support.

**Reply by Lausbua** [reply]
Die 192.168.178.74 ist ganz bestimmt die IP vom Shelly Plug S?
Dein Shelly Plus 1PM hat nämlich eine andere URL zum Auslesen vom Meter, und quittiert dann denn Versuch mit der URL vom Shelly Plug S eventuell mit Not Found.

Habe selbst einen Plug S im Einsatz, die URL so wie oben eingetragen funktioniert normalerweise einwandfrei.

![image](https://user-images.githubusercontent.com/3309871/197053127-b90b2aa0-36c9-438e-8368-cc4be46b9283.png)

Aktuellste Firmware auf dem Shelly Plug S? Benutzername/Passwort Authentifizierung am Webinterface nicht aktiv?

**Reply by masterbillRE11** [reply]
Super - danke. Hatte keine Authentifizierung gesetzt - jetzt kann ich die /meter/0-Anzeige aufrufen.
![image](https://user-images.githubusercontent.com/88536410/197279092-a1fb6ee2-8a4e-429d-8494-a84e54695680.png)

Leider wird die Leistung noch nicht im SHM 2.0 angezeigt. Unter Average steht noch "0"

![image](https://user-images.githubusercontent.com/88536410/197279386-cefc7430-a071-4482-84d2-2a61c9fbe370.png)

**Reply by masterbillRE11** [reply]
![image](https://user-images.githubusercontent.com/88536410/197282142-2440c188-036d-4b22-bae0-e8fc342d5162.png)
SAE 3
Method = GET? 
Habe ich hier alle Einstellungen richtig vorgenommen?

**Reply by camueller** [reply]
Warum schaust Du nicht mal ins Log? Dafür gibt es das ...
Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Logging_DE.md und https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/HttpMeter_DE.md#log

**Reply by Lausbua** [reply]
Ja, habe meinen Shelly Plug S genauso konfiguriert und ich habe im Sunny Portal zu meinen Geräten den Verbrauch.
Sowie bei einem gerade aktiven Verbraucher unter der SEMP Url einen AveragePower.

![image](https://user-images.githubusercontent.com/3309871/197284044-1d9ea817-333c-4e28-a894-2578e1f2f7fa.png)


![image](https://user-images.githubusercontent.com/3309871/197283845-c9b091a6-f4f4-41b9-84b2-3d4c98c9c0c0.png)

**Comment by masterbillRE11** [comment]


**Reply by camueller** [reply]
Irgendwie hast Du das Konzept von Gerät, Zähler und Schalter noch nicht verstanden. Der Shelly kann als Zähler und Schalter eingesetzt werden, aber Gerät muss das sein, was am Shelly dranhängt.

**Reply by masterbillRE11** [reply]
Scheint so :-) Danke. 
Als Gerät soll die TV-Bank gesteuert bzw. meinem Fall lediglich als Verbrauchsmesser im SHM aufgenommen werden. 
Wie könnte ich dieses Projekt realisieren?

**Reply by camueller** [reply]
In den Felder des Gerätes solltest Du das dann entsprechend eintragen, z.B. Hersteller "Diverse", Bezeichnung "TV-Bank", ...

An Zähler/Schalter ändert sich ja nichts dadurch.

**Reply by masterbillRE11** [reply]
done - die Fragestellung lt. dann nach wie vor, warum der erfolgreich im WLAN eingebundene Shelly keine Werte an den SHM weiterleitet.

**Reply by camueller** [reply]
Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md

**Reply by masterbillRE11** [reply]
SEMP2 > SAE3 > HTTP > Anpassungen Shelly notwendig? auf Basis des o. g. Screenshots

**Reply by masterbillRE11** [reply]
![image](https://user-images.githubusercontent.com/88536410/196950078-38ccfb96-cc51-412e-bf1e-fabeb7edba2b.png)
